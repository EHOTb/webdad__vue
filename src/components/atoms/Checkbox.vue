<template>
  <div :class="$style.items">
    <label :class="$style.item">
      <input type="checkbox" :class="$style.custom" :checked="checked" />
      <span :class="$style.label" />
    </label>
  </div>
</template>

<script>
export default {
  name: "Checkbox",
  props: { checked: Boolean },
};
</script>

<style lang="scss" module>
@import "../../assets/styles/var.scss";

.items {
  display: flex;
  flex-direction: column;
  align-items: center;
}
.item {
  display: flex;
  text-align: center;
  .custom {
    position: absolute;
    opacity: 0;
    z-index: -1;
  }
  .custom + span::before {
    content: "";
    display: inline-block;
    width: 1.5rem;
    height: 1.5rem;
    flex-shrink: 0;
    flex-grow: 0;
    border: 0.15rem solid $grad-sec-col;
    background-repeat: no-repeat;
    background-position: center center;
    border-radius: 0.25rem;
    margin-right: 0.5rem;
  }
  .custom:checked + span::before {
    background-color: $grad-sec-col;
    background-image: url("../../assets/ok.svg");
    background-repeat: no-repeat;
    background-position: 40% 40%;
  }
  .custom + span {
    display: inline-flex;
    align-items: center;
    user-select: none;
  }
}
</style>
