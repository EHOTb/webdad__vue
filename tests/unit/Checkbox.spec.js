import { mount } from "@vue/test-utils";
import Checkbox from "../../src/components/atoms/Checkbox";

describe("Компонент Checkbox", () => {
  it("Snapshot Checkbox", () => {
    const wrapper = mount(Checkbox);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("emit события Checkbox", () => {
    const wrapper = mount(Checkbox);
    wrapper.find("input").trigger("click");
    expect(wrapper.emitted("checkCheckbox")).toBeTruthy();
  });
  it("props", () => {
    const wrapper = mount(Checkbox, {
      propsData: {
        isChecked: true,
      },
    });
    expect(wrapper.find("input").element.checked).toBeTruthy();
  });
});
