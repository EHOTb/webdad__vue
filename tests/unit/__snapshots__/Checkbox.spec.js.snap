// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Checkbox Snapshot Checkbox 1`] = `
<label
  class="item"
>
  <input
    class="custom"
    type="checkbox"
  />
   
  <span />
</label>
`;
